Add edg-g-imx8mp by TechNexion

iashchepkov opened the pull request:
This patch is based on pico-imx8mm

Passed tests:
1. build
2. boot
3. connect to hosted.mender.io
4. firmware upgrade

TheYoctoJester commented:
Hi @iashchepkov,

thanks a lot for your submission. A couple of minor things that need fixing before I can merge:

1) can you please add your sign off?
2) this seems to add a new board, can you please also update the Readme accordingly so people can find it more easily?

Nice to have add-ons would be, but not merge-blocking:
- consider moving forward to `dunfell`, as the `zeus` release is end of life
- if there is a board support article on hub.mender.io that covers the already existing board, consider extending it or adding a second one in order to give the board more visibility.

@mirzak any thought on your side?

iashchepkov commented:
@TheYoctoJester thank you for the feedback!
I updated the PR: signed all commits and added board name to the README.md

Like PICO-PI-IMX8M-MINI it depends on TechNexion Yocto layer, which doesn't have `dunfell` branch.

mirzak commented on meta-mender-tn-imx-bsp/recipes-bsp/u-boot/u-boot-tn-imx/edm-g-imx8mp/0001-configs-edm-g-imx8mp-add-Mender-support.patch:
```diff
@@ -24,10 +24,11 @@ index 5d55972019..c736004c34 100644
 +CONFIG_SYS_REDUNDAND_ENVIRONMENT=y
 +CONFIG_BOOTCOUNT_LIMIT=y
 +CONFIG_BOOTCOUNT_ENV=y
++CONFIG_BOOTCOUNT_BOOTLIMIT=3
```
Why `CONFIG_BOOTCOUNT_BOOTLIMIT=3`? This not something that is typically needed. I think the default is 1?

iashchepkov replied:
I tested with images imx-* images. This images has a service that is trying to resize the root partition and reboots afterwards.
U-boot counted this reboot as fail and booted old firmware.
With bootlimit = 3 I was able to successfully upgrade firmware with mender.
I belive bootlimit =2 would work too but I noticed that for other boards bootlimit is set to 3.

To get imx-* images fully supported by mender, the resizing script should be remove, which is not part of this repository.

TheYoctoJester replied:
Very good catch @mirzak!

But this is quite specific and, for a user, unexpected behaviour then. This certainly mandates a board specific note in the Readme then.

mirzak replied:
> I tested with images imx-* images. This images has a service that is trying to resize the root partition and reboots afterwards.

I think disabling these scripts would the correct approach. Is it not possible with a bbappend here?

I think setting `bootlimit=3` will break rollback functionality of the Mender client as it expects the bootloader to perform a rollback if it reboots due to failure in e.g committing. @TheYoctoJester , maybe you can verify with someone from the client team.

kacf replied:
Yes, bootlimit has to be 1, otherwise the client may misbehave.

iashchepkov replied:
Thank you all for the feedback and the catch!

I will come back once I would be able to boot and upgrade without `bootlimit=3`.

TheYoctoJester replied:
Thanks! I also think that disabling such a script via an append should not hard to add.

mirzak reviewed (commented):
Think you covered most of it @TheYoctoJester. Had one question regarding bootlimit.

Agree that this would be very nice,

> if there is a board support article on hub.mender.io that covers the already existing board, consider extending it or adding a second one in order to give the board more visibility.

Then you can also add the link to the README.

iashchepkov commented:
@mirzak I would like to create a topic on hub.mender.io but unfortunately I'm very limited in time for this project :(

We discussed this patch with TechNexion by email and they know about this PR, I hope they are interested in this feature too and can provide public materials for their customers.

mirzak commented:
> @mirzak I would like to create a topic on hub.mender.io but unfortunately I'm very limited in time for this project :(

Understandable :+1:

iashchepkov commented:
@TheYoctoJester no worries, thank you! I hope you feel well!
